Upload/download/list files fails with irods setup where provider and consumer are split in two nodes

This ticket system is to report bugs and feature requests. For support, visit the [help page](https://cyberduck.io/help/) first.

 * If you have trouble connecting to a server or your login credentials are not valid, try to resolve the issue with the assistance of your hosting service provider. Any such ticket will be closed with a resolution of _thirdparty_.

**Describe the bug**
I need to connect CyberDuck to an irods setup where the irods provider (icat) and consumer (resource) servers live in two different nodes. When I connect to the provider node with CyberDuck, I can list the files, but upload/downloads of files hosted in the consumer node fail. If I connect directly to the consumer node,  authentication succeds but it does not lits any contents and upload/download doesn't work. 

I see this critical error on the irods server logs in both cases:

```
{"log_category":"server","log_level":"critical","log_message":" 0# 0x00007F33E869C06C\n 1# 0x00007F33E703EBF0\n 2# 0x00007F33E7AF4996\n 3# 0x00007F33E7AF4EF7\n 4# 0x00007F33DFDE0BB9\n 5# 0x00007F33DFDE88E5\n 6# 0x00007F33DFDF5522\n 7# 0x00007F33E8305AD3\n 8# 0x00007F33E830595F\n 9# 0x00007F33E82E8338\n10# 0x00007F33E82D69A6\n11# 0x00007F33DFDE94B7\n12# 0x00007F33DFDF56AA\n13# 0x00007F33E830869E\n14# 0x00007F33E83084AA\n15# 0x00007F33E82EAF36\n16# 0x00007F33E82D9E48\n17# 0x00007F33E82B029A\n18# 0x00007F33E82AFE3B\n19# 0x00007F33DFDADD4E\n20# 0x00007F33DFDAD6B5\n21# 0x00007F33DFDAD42C\n22# 0x00007F33E8248F70\n23# 0x00007F33E82477B0\n24# 0x00007F33E86AC4D9\n25# 0x00007F33E86C694A\n26# 0x00007F33E83C132F\n27# 0x00007F33E86D4746\n28# 0x00007F33E86D4D16\n29# 0x00007F33E86D5B36\n30# 0x00007F33E83F20A6\n31# 0x00007F33E85D08E3\n32# 0x00007F33E85D0825\n33# 0x00007F33E84E1D75\n34# 0x00007F33E86CC6BA\n35# 0x00007F33E86CE7B8\n36# 0x000055C0E3D24C07\n37# 0x000055C0E3D22362\n38# 0x000055C0E3D21135\n39# 0x00007F33E70295D0\n40# 0x00007F33E7029680\n41# 0x000055C0E3D1D2E5\n","server_host":"NOTME","server_pid":3250741,"server_timestamp":"2026-06-01T14:54:56.128Z","server_type":"server","server_zone":"NOTME","stacktrace_agent_pid":"3278770","stacktrace_timestamp_epoch_milliseconds":"619","stacktrace_timestamp_epoch_seconds":"1780325695","stacktrace_timestamp_utc":"2026-06-01T14:54:55.619Z"}
```

**Expected behavior**
When I connect to either provider or consumer via regular icommands, I can perform any operation successfully. I'm wondering if with CyberDuck the setup I've described is valid.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 9.4.1

Found these errors when the upload fails:

```
2026-06-02 12:59:44,980 [background-6] DEBUG ch.cyberduck.core.threading.BackgroundCallable - Canceled running background task WorkerBackgroundAction{worker=ConcurrentTransferWorker{source=DefaultSessionPool{bookmark=Host{protocol=Profile{parent=irods, vendor=iRODS Consortium, description=iRODS (Integrated Rule-Oriented Data System), image=null}, region='null', port=1247, hostname='xxxxxxxxxxxxxx', credentials=Credentials{user='xxxxxxxxxxxxxxxxxx', password='********', tokens='TemporaryAccessTokens{accessKeyId='', secretAccessKey='', sessionToken='', expiryInMilliseconds=-1}', oauth='OAuthTokens{accessToken='', refreshToken='', idToken='', expiryInMilliseconds=-1}', identity=null, properties={}}, uuid='19696bf4-d006-4d0c-9846-4e6fe9a11dbf', nickname='null', defaultpath='null', workdir=Path{path='xxxxxxxxxxxxxxxxxxx', type=[directory]}, custom=null, labels=null}, idle=1, active=0, waiters=0}, destination=ch.cyberduck.core.pool.SessionPool$DisconnectedSessionPool@721596, pool=java.util.concurrent.ExecutorCompletionService@3a628d1}}
java.lang.Exception
	at ch.cyberduck.core.threading.BackgroundCallable.<init>(BackgroundCallable.java:38)
	at ch.cyberduck.core.threading.DefaultBackgroundExecutor.execute(DefaultBackgroundExecutor.java:72)
	at ch.cyberduck.core.AbstractController.background(AbstractController.java:72)
	at cli.Ch.Cyberduck.Ui.Controller.TransferController.StartTransfer(Unknown Source)
	at cli.Ch.Cyberduck.Ui.Controller.BrowserController.UploadDroppedPath(Unknown Source)
	at cli.Ch.Cyberduck.Ui.Controller.BrowserController.View_BrowserDropped(Unknown Source)
	at cli.Ch.Cyberduck.Ui.Winforms.BrowserForm$ExpandingBrowserDropSink.OnDropped(Unknown Source)
	at cli.BrightIdeasSoftware.SimpleDropSink.TriggerDroppedEvent(Unknown Source)
	at cli.BrightIdeasSoftware.SimpleDropSink.Drop(Unknown Source)
	at cli.BrightIdeasSoftware.ObjectListView.OnDragDrop(Unknown Source)
	at cli.System.Windows.Forms.DropTarget.System.Windows.Forms.UnsafeNativeMethods.IOleDropTarget.OleDrop(Unknown Source)
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source)
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source)
	at cli.System.Windows.Forms.Application$ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(Unknown Source)
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoopInner(Unknown Source)
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoop(Unknown Source)
	at cli.Ch.Cyberduck.Ui.Program.Main(Unknown Source)
Caused by: ch.cyberduck.core.exception.TransferCanceledException: Unknown
	at ch.cyberduck.core.worker.AbstractTransferWorker$3.transferSegment(AbstractTransferWorker.java:471)
	at ch.cyberduck.core.worker.AbstractTransferWorker$3.call(AbstractTransferWorker.java:356)
	at ch.cyberduck.core.worker.AbstractTransferWorker$3.call(AbstractTransferWorker.java:346)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:512)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:625)
	at ch.cyberduck.core.threading.NamedThreadFactory$1.run(NamedThreadFactory.java:57)
	at java.lang.Thread.run(Thread.java:980)
Caused by: ch.cyberduck.core.exception.ConnectionCanceledException: Error
	at ch.cyberduck.core.AbstractExceptionMappingService.wrap(AbstractExceptionMappingService.java:122)
	at ch.cyberduck.core.worker.DefaultExceptionMappingService.map(DefaultExceptionMappingService.java:42)
	at ch.cyberduck.core.irods.IRODSUploadFeature.upload(IRODSUploadFeature.java:263)
	at ch.cyberduck.core.irods.IRODSUploadFeature.upload(IRODSUploadFeature.java:55)
	at ch.cyberduck.core.vault.registry.VaultRegistryUploadFeature.upload(VaultRegistryUploadFeature.java:46)
	at ch.cyberduck.core.transfer.UploadTransfer.transfer(UploadTransfer.java:327)
	at ch.cyberduck.core.worker.AbstractTransferWorker$3.transferSegment(AbstractTransferWorker.java:405)
	... 9 more
Caused by: java.nio.BufferUnderflowException
	at java.nio.Buffer.nextGetIndex(Buffer.java:510)
	at java.nio.HeapByteBuffer.getInt(HeapByteBuffer.java:375)
	at org.irods.irods4j.low_level.network.Network.readMsgHeader_PI(Network.java:55)
	at org.irods.irods4j.low_level.api.IRODSApi.receiveServerResponse(IRODSApi.java:211)
	at org.irods.irods4j.low_level.api.IRODSApi.rcReplicaOpen(IRODSApi.java:664)
	at org.irods.irods4j.high_level.io.IRODSDataObjectStream.openImpl(IRODSDataObjectStream.java:423)
	at org.irods.irods4j.high_level.io.IRODSDataObjectStream.open(IRODSDataObjectStream.java:121)
	at org.irods.irods4j.high_level.io.IRODSDataObjectOutputStream.open(IRODSDataObjectOutputStream.java:162)
	at org.irods.irods4j.high_level.io.IRODSDataObjectOutputStream.<init>(IRODSDataObjectOutputStream.java:69)
	at ch.cyberduck.core.irods.IRODSUploadFeature.upload(IRODSUploadFeature.java:110)
	... 13 more
```

--- a/irods/pom.xml
+++ b/irods/pom.xml
@@ -27,7 +27,7 @@
         <dependency>
             <groupId>org.irods</groupId>
             <artifactId>irods4j</artifactId>
-            <version>0.6.0-java8</version>
+            <version>0.7.0-java8</version>
         </dependency>
         <dependency>
             <groupId>ch.cyberduck</groupId>

